controllers/metallb_controller.go

```diff
@@ -42,9 +42,27 @@ type MetallbReconciler struct {
 
 var ManifestPath = "./bindata"
 
-//TODO Adjust with relevant permissions only
-
-// +kubebuilder:rbac:groups=*,resources=*,verbs=get;list;watch;create;update;patch;delete
+// Namespace Scoped
+// +kubebuilder:rbac:groups=apps,namespace=metallb-system,resources=deployments;daemonsets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,namespace=metallb-system,resources=roles;rolebindings,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",namespace=metallb-system,resources=serviceaccounts,verbs=get;list;watch;create;update;patch;delete
+
+// Cluster Scoped
+// +kubebuilder:rbac:groups=metallb.io,resources=metallbs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=metallb.io,resources=metallbs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;clusterrolebindings,verbs=get;list;watch;create;update;patch;delete
```

Comment:
This is giving us superpowers I think, and we need it because we need to create the permissions in the metallb manifests.
Is there another way to achieve that? What kind of permissions do the created pods get if we don't add this?

Reply:
I didn't find a different way to achieve that and I guess the metallb pods are not worth much without their clusterroles.
I seperated between create and the other verbs to make it more specific, but I'm not sure how much value it adds. (create can't be restricted by resourceName https://github.com/kubernetes/kubernetes/issues/80295#issuecomment-512874100 )